### PR TITLE
Update all InstructorFeedbackResults*.js to ES6 #6771

### LINKED ISSUE #6771: Update all InstructorFeedbackResults*.js to ES6
No functionality changes. Tentatively still relying on the old common.js mess for now, until #6767 (or a similar PR) is merged.

### PR DESCRIPTION
**Pull request by thenaesh**
Fixes #6771 but still WIP. More commits for the other files incoming.

### CODE REVIEW

**Comment by thenaesh** [comment]
@wkurniawan07 @Neurrone Ready for review.

Feel free to ignore the ES6 and just check that all the compiled JS files are nearly identical to the originals. Changes on top of `eslint --fix` are minimal. :stuck_out_tongue:

File: src/main/webapp/js/instructorFeedbackResults.es6 [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+          toggleSingleCollapse:false
+*/
+
+$(document).ready(() => {
+    const participantPanelType = 'div.panel.panel-primary,div.panel.panel-default';
+
+    $('a[id^="collapse-panels-button-section-"]').on('click', (e) => {
+        const isGroupByTeam = document.getElementById('frgroupbyteam').checked;
+        const childPanelType = isGroupByTeam ? 'div.panel.panel-warning' : participantPanelType;
+        const panels = $(e.currentTarget).closest('.panel-success')
```
Comment (thenaesh) [inline src/main/webapp/js/instructorFeedbackResults.es6]: Tell me if you guys want me to revert the `$(e.currentTarget)` part back to `$(this)` and change the arrow function back into a normal one.

**Comment by wkurniawan07** [comment]
@thenaesh let's remove the transpiled ES5 files?

**Comment by thenaesh** [comment]
@wkurniawan07 Done! Travis gives green now (hopefully removing the transpiled ES5 doesn't make it red again :new_moon_with_face:). 

@Neurrone Ready for review! xD

File: src/main/webapp/js/instructorFeedbackResults.es6 [hunk 2]
```diff
@@ -1,40 +1,50 @@
...
+/* global selectElementContents:false,
+          bindCollapseEvents:false,
+          bindPublishButtons:false,
+          bindUnpublishButtons:false,
+          setStatusMessage:false,
+          showSingleCollapse:false,
+          hideSingleCollapse:false,
+          toggleSingleCollapse:false
+*/
+/* eslint-disable no-use-before-define */
```
Comment (Neurrone) [inline src/main/webapp/js/instructorFeedbackResults.es6]: Shouldn't be necessary, since the .yaml config already turns it off globally

File: src/main/webapp/js/instructorFeedbackResults.es6 [hunk 3]
```diff
@@ -1,40 +1,50 @@
...
-    });
+    $('#results-search-box').keyup(updateResultsFilter);
 
     // prevent submitting form when enter is pressed.
-    $('#results-search-box').keypress(function(e) {
+    $('#results-search-box').keypress((e) => {
         if (e.which === 13) {
             return false;
         }
+        return true;
```
Comment (Neurrone) [inline src/main/webapp/js/instructorFeedbackResults.es6]: Is this needed? Returning true is the same as not returning anything (nothing is done).

File: src/main/webapp/js/instructorFeedbackResults.es6 [hunk 4]
```diff
@@ -1,40 +1,50 @@
...
-    $('#results-search-box').keyup(function() {
-        updateResultsFilter();
-    });
+    $('#results-search-box').keyup(updateResultsFilter);
 
     // prevent submitting form when enter is pressed.
-    $('#results-search-box').keypress(function(e) {
+    $('#results-search-box').keypress((e) => {
         if (e.which === 13) {
             return false;
```
Comment (Neurrone) [inline src/main/webapp/js/instructorFeedbackResults.es6]: From the slack discussion, this is not recommended. Just flagging it for fixing in a future PR.

File: src/main/webapp/js/instructorFeedbackResults.es6 [hunk 5]
```diff
@@ -69,39 +79,39 @@ $(document).ready(function() {
...
+                    const table = data.sessionResultsHtmlTableAsString;
+                    content.html(`<small>${table}</small>`);
                     ajaxStatus.html(data.ajaxStatus);
                 }
                 setStatusMessage(data.statusForAjax);
             }, 500);
-        }
+        },
     });
 }
```
Comment (Neurrone) [inline src/main/webapp/js/instructorFeedbackResults.es6]: Another one for future PR: it won't let me insert this comment inside the actual function, but showHideStats should be renamed to e.g toggleStatsDisplay.

File: src/main/webapp/js/instructorFeedbackResults.es6 [hunk 6]
```diff
@@ -206,59 +216,57 @@ function updateStatsCheckBox() {
...
-        var heading = $(panels[i]).children('.panel-heading');
-        var bodyCollapse = $(panels[i]).children('.panel-collapse');
+    let numPanels = nPanels;
+    for (let i = 0; i < panels.length; i += 1) {
+        const heading = $(panels[i]).children('.panel-heading');
+        const bodyCollapse = $(panels[i]).children('.panel-collapse');
         if (heading.length !== 0 && bodyCollapse.length !== 0) {
-            numPanels++;
+            numPanels += 1;
             // $(heading[0]).attr('data-toggle', 'collapse');
```
Comment (Neurrone) [inline src/main/webapp/js/instructorFeedbackResults.es6]: Dead/unused code?

File: src/main/webapp/js/instructorFeedbackResultsAjaxByGQR.es6 [hunk 7]
```diff
@@ -1,39 +1,49 @@
...
+          isEmptySection:false,
+          removeSection:false,
+          bindDefaultImageIfMissing:false,
+          bindStudentPhotoLink:false,
+          bindStudentPhotoHoverLink:false,
+          bindCollapseEvents:false,
+          toggleSingleCollapse:false,
+          showHideStats:false
+*/
+/* eslint-disable no-use-before-define */
```
Comment (Neurrone) [inline src/main/webapp/js/instructorFeedbackResultsAjaxByGQR.es6]: Here too, and I guess other files as well.

File: src/main/webapp/js/instructorFeedbackResults.es6 (hunk 3 above)
Comment (thenaesh) [inline src/main/webapp/js/instructorFeedbackResults.es6]: This one is because ESLint screamed at me. I'll check if it still screams at me if I remove it.

File: src/main/webapp/js/instructorFeedbackResults.es6 (hunk 2 above)
Comment (thenaesh) [inline src/main/webapp/js/instructorFeedbackResults.es6]: Oh yea, thanks for pointing this out.

File: src/main/webapp/js/instructorFeedbackResults.es6 (hunk 4 above)
Comment (thenaesh) [inline src/main/webapp/js/instructorFeedbackResults.es6]: Why? In this situation there is no difference. Or is it because it's an unnecessary change (which it technically is xD).

File: src/main/webapp/js/instructorFeedbackResults.es6 (hunk 3 above)
Comment (thenaesh) [inline src/main/webapp/js/instructorFeedbackResults.es6]: Ok, according to @wkurniawan07's suggestion on Slack, I'll use `e.preventDefault()` for this. That kills two birds with one stone: readability and ESLint.

File: src/main/webapp/js/instructorFeedbackResults.es6 (hunk 5 above)
Comment (thenaesh) [inline src/main/webapp/js/instructorFeedbackResults.es6]: This one really probably for a future PR.

File: src/main/webapp/js/instructorFeedbackResults.es6 (hunk 4 above)
Comment (Neurrone) [inline src/main/webapp/js/instructorFeedbackResults.es6]: Oh its because return false does both an e.preventDefault() and e.stopPropagation()

File: src/main/webapp/js/instructorFeedbackResults.es6 (hunk 6 above)
Comment (thenaesh) [inline src/main/webapp/js/instructorFeedbackResults.es6]: What on earth... that needs to go. Thanks for the catch!

**Comment by thenaesh** [comment]
@Neurrone Changes made! :)
